src/jailer/src/resource_limits.rs

```diff
@@ -98,6 +98,24 @@ impl ResourceLimits {
         // Set limit on number of file descriptors.
         ResourceLimits::set_limit(Resource::RlimitNoFile, self.no_file)?;
 
+        // Resize the file descriptor table to its maximal possible size, to ensure that
+        // firecracker will not need to reallocate it later. If the file descriptor table
+        // needs to be reallocated (which by default happens once more than 64 fds exist,
+        // something that happens for reasonably complex microvms due to each device using
+        // a multitude of eventfds), this can incur a significant performance impact (it
+        // was responsible for a 30ms-70ms impact on snapshot restore times).
+        if self.no_file > 64 {
```

Comment:
64 here is driven from a constant in the kernel. I think we should not rely to that in case it changes in the future. Do you see any issue with removing the `if` statement all together?

Reply:
What's going to happen if ` self.no_file as libc::c_int - 1` already exists? I don't think we'd like to replace/close it.

Reply:
We could try to `dup2` `self.no_file as libc::c_int` instead. Nothing should be using that, right?

Reply:
but that would be rejected by the jailer?

Reply:
doesn't jailer limit the number of 'opened' fds? When we do the 'dup2' we just ask for 1 mode fd, but with big number. So this is a +1 -1 situation. Technically we can resize table to any size. It is just only `no_file` number of fds can be open at the same time.

Reply:
I think what Egor says is true. If it isn't, we could just apply this hack before setting the limit, right?

Reply:
> 64 here is driven from a constant in the kernel. I think we should not rely to that in case it changes in the future. Do you see any issue with removing the `if` statement all together?

We need some check here because if `no_file = 3` then `dup2` would fail (as stdin, stdout, stderr are already 3 open fds). We actually have a test for this. So we can change the check to 3 (I picked 64 because it doesnt makes sense to do it if no_file=64, but I can see how hardcoding kernel details is not nice, so will remove)